framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -136,6 +139,11 @@ protected CFAbstractTransfer(
     this.analysis = analysis;
     this.sequentialSemantics =
         !(forceConcurrentSemantics || analysis.checker.hasOption("concurrentSemantics"));
+    this.resourceLeakChecker =
+        analysis
+            .checker
+            .getUpstreamCheckerNames()
+            .contains("org.checkerframework.checker.resourceleak.ResourceLeakChecker");
```

Comment:
Putting this in `CFAbstractTransfer` seems like too much coupling, to me. If you're going to use this implementation strategy (which I'd prefer we replace; see my comment on the first use of the `isResourceLeakCheckerEnabled` above), this logic should be in some Resource Leak Checker class (maybe as a static method that takes a `BaseTypeChecker`?)